xmake/rules/c++/modules/build_modulefiles.lua

```diff
@@ -125,14 +125,13 @@ function main(target, sourcebatch, opt)
 
     -- do compile
     local compinst = compiler.load("cxx")
```

Comment:
please help me to modify this line, use target:tool to get compiler name

```lua
        local _, toolname = target:tool("cxx")
        if toolname == "clang" then -- gcc, cl
```

Reply:
I will need to revert to version 2.3.6 that is the last I was able to properly compile xmake. You are right about supporting older versions. I will change the code and present a new pull request.

Reply:
I made the changes you suggested and xmake builds my project with modules just fine. Please close this pull request and I will create a new one with the more generic code to build modules, including support for previous compiler versions with `-fmodules-ts`.

Reply:
ok